Automattic/sensei
# [POC ]Add learning mode v2

## Proposed change
* Introduce a learning mode v2
* Split the learning mode template into patterns
* Load specific CSS files only when the new learning mode template is used. 


## Testing instructions
* Go to this branch
* Run npm run start
* Go to the Sensei > Settings > Appearance 
* Enable the `Default Template V2" 
* Go to the Site Editor
* Edit. the new Lesson (Learning Mode - Default V2)
* Check it was rendered with an ugly border red < it means we are using the specialized CSS.


### Context
The motivation of this POC is to test if we can have a safe and efficient way to improve the current learning mode templates safely, without being worried if it will break the existing courses.  For this PR, I am only checking the feature flag, but we can use any kind of logic we want to manage when the new template will be available. E.g. WordPress version.

**1) Add a Learning mode v2 template based in a feature flag**
It enables us to evolve the learning mode without being worried about introducing breaking changes, so for example, if we need to create a v2 version from an existing block, we can use the new template and keep the old one with the old template. 

Another benefit is enabling our users to adopt the reviewed learning mode gradually and safely, avoiding drastic changes.

**2) Split the default learning mode template into patterns** 
It enables us to work better in parallel because we can split the cards by pattern, avoiding conflicts. 

**3) Load new CSS that only will be applied to the V2 version.***
In case we need to add custom CSS to fix the layout,  we can do it easily directly on the V2 file, without being worried it will break the existing version.

**4) Refactor the header to use the WordPress regular layout block, instead of the custom block to manage the layout .***


### Detailed explanation:
I am adding [here](https://github.com/Automattic/sensei/blob/9ed427ece0875e9a2ca341abd210c0e2a8e5654c/includes/course-theme/class-sensei-course-theme-template-selection.php#L159-L181)  a new template to the Sensei > Settings > Template Selection.

I am adding [Here](https://github.com/Automattic/sensei/blob/9ed427ece0875e9a2ca341abd210c0e2a8e5654c/includes/block-patterns/class-sensei-block-patterns.php#L122-L127)  the learning mode patterns.
 
### Extra notes:
* We do not necessarily need to use all suggestions together, we can use the V2 template + the pattern idea and just reuse the current layout blocks.

<img width="1399" alt="Screenshot 2023-04-04 at 10 31 53" src="https://user-images.githubusercontent.com/38718/229750271-8cc44a3f-afeb-4e15-9415-45a5d97b6732.png">
